MdeModulePkg/Bus/Usb/UsbBusDxe/UsbEnumer.c

```diff
@@ -995,7 +995,9 @@ UsbEnumeratePort (
     // Now, new device connected, enumerate and configure the device
     //
     DEBUG ((DEBUG_INFO, "UsbEnumeratePort: new device connected at port %d\n", Port));
-    if (USB_BIT_IS_SET (PortState.PortChangeStatus, USB_PORT_STAT_C_RESET)) {
+    if (USB_BIT_IS_SET (PortState.PortChangeStatus, USB_PORT_STAT_C_RESET) &&
+        (Status != EFI_DEVICE_ERROR))     // MU_CHANGE
+    {

```

Comment:
This doesn't look right. It looks like we'd be forcing the first branch, and never taking the second branch. The only thing that sets Status seems to be the GetPortStatus at the top of the function, and if Status were EFI_DEVICE_ERROR we would have already returned. Therefore, Status == EFI_SUCCESS and this will always return true, right?

Reply:
Ah, that's true my initial revision had this on line 916, which was omitted while rebasing to 202111, will update it. 
```C
if (EFI_ERROR (Status)) && (Status != EFI_DEVICE_ERROR))
```

Reply:
Fixed in the latest. We'll now not return immediately on `EFI_DEVICE_ERROR` but instead try a reset if there is a device connected. If not, Status is returned at the end of the function anyway so `EFI_DEVICE_ERROR` will still be returned but it will happen later.